pkg/client/helpers.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func Generate(ownershipLevel, action string, resourceHierarchy ...string) func(r *http.Request) string {
+	if len(resourceHierarchy) == 0 {
+		resourceHierarchy = []string{"*"}
+	}
+
+	return func(r *http.Request) string {
+		if wi := Get(r); wi != nil {
+			return fmt.Sprintf(
+				"%s::%s::%s::%s",
+				wi.GetServiceName(),
+				ownershipLevel,
+				action,
+				strings.Join(resourceHierarchy, "::"),
+			)
+		}
+
+		return ""
+	}
+}
+
+func GenerateInfo() func(r *http.Request) string {
```

Comment:
What this is used for?

Reply:
This generate an empty permission that can be used to only get information about the user without career of permission. Example: You want to know the services account email